src/libraries/Common/src/System/Net/Security/SslClientAuthenticationOptionsExtensions.cs

```diff
@@ -2,22 +2,76 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Collections.Generic;
+#if DEBUG
+using System.Collections;
+using System.Diagnostics;
+using System.Reflection;
+#endif
+
 namespace System.Net.Security
 {
     internal static class SslClientAuthenticationOptionsExtensions
     {
-        public static SslClientAuthenticationOptions ShallowClone(this SslClientAuthenticationOptions options) =>
-            new SslClientAuthenticationOptions()
+        public static SslClientAuthenticationOptions ShallowClone(this SslClientAuthenticationOptions options)
+        {
+            var clone = new SslClientAuthenticationOptions()
             {
                 AllowRenegotiation = options.AllowRenegotiation,
-                ApplicationProtocols = options.ApplicationProtocols,
+                ApplicationProtocols = options.ApplicationProtocols != null ? new List<SslApplicationProtocol>(options.ApplicationProtocols) : null,
                 CertificateRevocationCheckMode = options.CertificateRevocationCheckMode,
+                CipherSuitesPolicy = options.CipherSuitesPolicy,
                 ClientCertificates = options.ClientCertificates,
                 EnabledSslProtocols = options.EnabledSslProtocols,
                 EncryptionPolicy = options.EncryptionPolicy,
                 LocalCertificateSelectionCallback = options.LocalCertificateSelectionCallback,
                 RemoteCertificateValidationCallback = options.RemoteCertificateValidationCallback,
                 TargetHost = options.TargetHost
             };
+
+#if DEBUG
+            // Try to detect if a property gets added that we're not copying correctly.
```

Comment:
It seems like this is something that could equally be done in a test ? (Using one more bit of reflection to get to the internal class)

Reply:
Also then maybe the code could be reused if there are other such places to check.

Reply:
There are probably quite a few tests (such as asserting method output is consistent with method input) that we could choose to write as `#if DEBUG` blocks but do not. I guess we do not because it's nice to have the tests together, it's easier to reuse the test code, it makes the product code less cluttered.

I guess it's a matter of taste, but this is quite a large assert block. I don't feel strongly though if you prefer it.

Reply:
I think this is better structured as an assert.  This is about validating the implementation is properly constructed.  In my mind it's akin to when we have debug asserts that validate we've presized a dictionary correctly.

There could be tests as well, e.g. to validate that setting a CipherSuitePolicy is correctly propagated.  But I wasn't sure how to do that in a reliable way across platforms.

Reply:
Fair enough